src/App.js

```diff
@@ -12,10 +13,12 @@ class App extends Component {
   render() {
     return (
       <Provider store={store}>
-        <MainNavbar/> 
+        
         <Router>
           <React.Fragment>
+            <MainNavbar/> 
             <Route exact path="/" component={MainPage} />
+            <Route exact path="/:slug/userpanel" component={UserPanel} /> 
```

Comment:
For user better is "username" not "slug" but I suggest a change route i.e. "/user/:username". That can help us avoid conflicts with switching a screen.